src/Components/Pages/Calculator.js

```diff
@@ -1,13 +1,208 @@
-import PropTypes from 'prop-types';
+import PropTypes from 'prop-types'
 import React from 'react'
+import WASMWrapper from '../../Backend/WASMWrapper'
+
+import CalculatorSettings from '../CalculatorHelpers/CalculatorSettings'
+import CalculatorInput from '../CalculatorHelpers/CalculatorInput'
+import CalculatorOutput from '../CalculatorHelpers/CalculatorOutput'
+
+import * as convert from 'convert-units'
+
+const Units = {
+    Inputs: {
+        MeVPerNucleum: 'MeV/nucl',
+        //        Mev: 'Mev',
+    },
+    Outputs: {
+        SmallScale: 'keV/μm',
+        LargeScale: 'MeV/cm',
+        MassStoppingPower: 'Mev*g/cm^2'
+
+    }
+}
+
+const OperationMode = {
+    Dynamic: 0,
+    Performance: 1,
+}
+
 
 class CalculatorComponent extends React.Component {
     static propTypes = {
         ready: PropTypes.bool.isRequired
     }
 
+    async componentDidMount() {
+        try {
+            const [programs, ions, materials] = await Promise.all([
+                this.wrapper.getPrograms(),
+                this.wrapper.getIons(this.state.program),
+                this.wrapper.getMaterials(this.state.program)
+            ])
+            const program = programs[0]
+            const material = materials[0]
+            const ion = ions[0]
+            this.setState({ programs, ions, materials, program, material, ion })
+        } catch (err) {
+            console.log(err)
+        }
+    }
+
+    async onProgramChange(newProgram) {
+        const programNumber = (Number)(newProgram.target.value)
+        try {
+            const [materials, ions] = await Promise.all([
+                this.wrapper.getMaterials(programNumber),
+                this.wrapper.getIons(programNumber)]
+            )
+            const material = materials[0]
+            const ion = ions[0]
+            const program = this.state.programs.find(prog => prog.code === programNumber)
+            this.setState({ materials, ions, program, material, ion })
+        } catch (err) {
+            console.log(err)
+        }
+    }
+
+    onChanges = {
+        onInputUnitChange: ({ target }) => {
+            const inputUnit = target.value
+            this.setState({ inputUnit })
+        },
+        onOutputUnitChange: ({ target }) => {
+            const outputUnit = target.value
+            //const result = this.enrichWithUnit(this.state.result, outputUnit)
```

Comment:
Why this is commented out ?

Reply:
There is no unit support for CSDA range for now (units will be calculated dynamically once we have the value)

Reply:
OK